src/components/PopupWithForm.js

```diff
@@ -20,13 +35,34 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this._form.addEventListener("submit", (event) => {
       event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this.close();
+      // this._handleFormSubmit(this._getInputValues());
+      const result = this._handleFormSubmit(this._getInputValues());
+      if (result instanceof Promise) {
+        result
+          .then(() => {
+            this.close();
+          })
+          .catch((error) => {
+            console.error("Error during confirmation:", error);
+          });
+      } else {
+        this.close();
```

Comment:
Please, delete the whole code from line 40 till 50 in `PopupWithForm`. It doesn't work. 


There should be only this


```js

  setEventListeners() {
    super.setEventListeners();
    this._form.addEventListener("submit", (event) => {
      event.preventDefault();
      this._handleFormSubmit(this._getInputValues());
    });
  }
```

And you don't need ` this.close();` here otherwise the modal closes immediately without waiting